content/doc/book/security/scripted.adoc

```diff
@@ -0,0 +1,70 @@
+---
+title: Scripted Security
+layout: section
+---
+ifdef::backend-html5[]
+:toc:
+ifdef::env-github[:imagesdir: ../resources]
+ifndef::env-github[:imagesdir: ../../resources]
+:hide-uri-scheme:
+endif::[]
+
+Apache Groovy scripts can be executed on the Jenkins controller and agents through various mechanisms including:
+
+* link:/doc/book/managing/script-console/[Script Console]
+* link:https://plugins.jenkins.io/groovy/[Groovy plugin] 
+when using the "Execute system Groovy script" step
+* link:https://plugins.jenkins.io/email-ext/[Extended Email plugin]
+* link:https://plugins.jenkins.io/groovy/[Job DSL plugin]
```

Comment:
Maybe it's better to structure this page more explicitly along the following lines to guide readers along a top-level view of the concepts involved?

1. Scripting capabilities are very powerful, but are potentially dangerous if mishandled.
2. The unrestricted Groovy-based Script Console is a powerful diagnostic tool and can only be used by admins. It has full access to all Jenkins internals and no restrictions.
3. Many plugins offer scripting capabilities (this list) to users without Overall/Administer permission, and they integrate with Script Security plugin to do this more safely.
4. Script Security plugin has a _sandbox_ for Groovy scripts and Pipelines, and _whole-script approval_ for all languages it supports.
5. The sandbox allows regular users to implement scripts safely, restricting access to potentially unsafe code based on an allowlist (default + admin customizable, but that is very complicated)
6. Whole-script approval lets users ask admins to approve entire scripts. This is generally discouraged.

Reply:
Thanks for this!  As you may have guessed, I did not understand this well but this clarifies things a lot!  I will rewrite.